extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/common/KubernetesPeonClient.java

```diff
@@ -120,22 +120,46 @@ public boolean deletePeonJob(K8sTaskId taskId)
     }
   }
 
-  public Optional<InputStream> getPeonLogs(K8sTaskId taskId)
+  public Optional<LogWatch> getPeonLogWatcher(K8sTaskId taskId)
   {
     KubernetesClient k8sClient = clientApi.getClient();
     try {
       LogWatch logWatch = k8sClient.batch()
+          .v1()
```

Comment:
adding this caused issues because the client (and hence the stream) gets closed before it's read to the requester.
current logic is no good either though because it leaves the KubernetesClient open, going to try to come up with another solution, hopefully that doesn't involve reading the entire stream into memory

Reply:
@suneet-s for now i'm gonna push a version of this that loads everything into memory again so this at least works properly.

I personally think it would be okay for us to keep a longer lived copy of KubernetesClient around somewhere to facilitate streaming but I'll have to think about that some more